include/envoy/upstream/upstream.h

```diff
@@ -650,13 +656,28 @@ struct ClusterCircuitBreakersStats {
   ALL_CLUSTER_CIRCUIT_BREAKERS_STATS(GENERATE_GAUGE_STRUCT, GENERATE_GAUGE_STRUCT)
 };
 
+/**
+ * Struct definition for cluster timeout budget stats. @see stats_macros.h
+ */
+struct ClusterRequestResponseSizeStats {
+  ALL_CLUSTER_REQUEST_RESPONSE_SIZE_STATS(GENERATE_HISTOGRAM_STRUCT)
+};
+
+using ClusterRequestResponseSizePtr = std::unique_ptr<ClusterRequestResponseSizeStats>;
+using ClusterRequestResponseSizeOptRef =
```

Comment:
is there is a reason we need to keep ptr and optRef types? would not be ptr enough?

Reply:
Based on discussion [here](https://github.com/envoyproxy/envoy/pull/11559#discussion_r446968854).
Also, If we return a unique_ptr<T>, then there is the issue with transfer of ownership which I don't think is intended.
If we return a pointer T* directly, the caller has to know if it is responsible for freeing the memory or someone else is. That's not something we are concerned with when it's a reference.

Reply:
got it